lib/Runtime/HostManager/HostManager.cpp

```diff
@@ -102,17 +103,20 @@ llvm::Error HostManager::addNetwork(std::unique_ptr<Module> module,
   auto partitioner = Partitioner(module.get(), deviceInfo, saturateHost);
   RETURN_IF_ERR(partitioner.Partition(cctx));
   auto nodeList = std::move(partitioner.getPartitionResult());
+
   if (cctx.precisionConfig.quantMode == QuantizationMode::Profile) {
-    // Check that all functions were not partitioned.
-    for (auto &network : nodeList) {
-      if (network.nodes.size() > 1) {
-        return MAKE_ERR(
-            GlowErr::ErrorCode::RUNTIME_ERROR,
-            "Failed to add network for profiling: Network was "
-            "partitioned, this is likely because the network was "
-            "larger than the configured memory of a single device manager.");
-      }
+    // For profiling, we use CPU backend. Overwrite Provisioner and Executor to
```

Comment:
I think we discussed in previous review rounds that it would be nice to have all the quantization specific logic in one place. And you suggested to use `Partitioner` for it. Or do we need to do this in the host manager because there are no easy ways to hide it all in the `Partitioner`?

Reply:
We leave the decision of which partition flow should be called to Partitioner. However, this is for Provisioner and Executor initialization, and this should be done in HM.